src/store.cc

```diff
@@ -772,6 +777,19 @@ StoreEntry::write (StoreIOBuffer writeBuffer)
     invokeHandlers();
 }
 
+void
+StoreEntry::writeData(StoreIOBuffer writeBuffer)
+{
+    assert(mem_obj);
+    debugs(20, 6, "offset " << writeBuffer.offset << " len " << writeBuffer.length);
+    auto &dataHdr = mem_obj->data_hdr;
+    assert (dataHdr.endOffset() || writeBuffer.offset == 0);
+    const auto oldSize = dataHdr.size();
+    assert(dataHdr.write(writeBuffer));
+    if (needsIdlePagesUpdating())
+        dataHdr.updateIdleNodes(oldSize);
```

Comment:
Why do we need to call this explicitly here? In other words, why cannot dataHdr.write() adjust the idle page counter as needed? Do we not trust its isIdle state??